### PR TITLE
Lookup: fix unnecessary rerender on clickoutside

### PR DESCRIPTION
**Pull request by brunottonurb**
Issue #220

I think the rerenders are triggered by enhanceWithClickOutside and the lack of a shouldComponentUpdate function.

I put in the shallow comparison func for props and state from the [legacy pureRenderMixin](https://developmentarc.gitbooks.io/react-indepth/content/life_cycle/update/using_should_component_update.html)

If this is an acceptable solution I will put `shallowCompare` into `utils`.

### CODE REVIEW

**Comment by brunottonurb** [comment]
@extronics I wasn't sure if I could do that because of

> Furthermore, React.PureComponent’s shouldComponentUpdate() skips prop updates for the whole component subtree. Make sure all the children components are also “pure”.

**Comment by extronics** [comment]
@brunottonurb should be no issue in this case because all children of Lookup are functional components (which are pure).

**Comment by brunottonurb** [comment]
After doing some fiddling around, I think using PureComponent is the way to go: The case where a rerender is triggered needlessly by clickOutside is when the prop `open` did not change. A custom `shouldComponentUpdate` function would essentially do the same thing as the PureComponent shallowCompare.

File: src/components/Lookup/Lookup.js [hunk 1]
```diff
@@ -1,10 +1,11 @@
-import React, { Component } from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import enhanceWithClickOutside from 'react-click-outside';
 import debounce from 'lodash.debounce';
 import omit from 'lodash.omit';
 
+// import shallowCompare from '../../utils';
```
Comment (fspoettel) [inline src/components/Lookup/Lookup.js]: Remove

File: src/components/Lookup/Lookup.js [hunk 2]
```diff
@@ -218,6 +219,45 @@ export class LookupRaw extends Component {
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
```
Comment (fspoettel) [inline src/components/Lookup/Lookup.js]: Remove

**Pull request merged**